dsalglib/include/array.h

```diff
@@ -26,9 +26,13 @@ namespace dsa
                     count = size;
                     capacity=size+spare;
                     objs = new type[capacity];
+                    if(!objs)
+                    	throw "Memory allocation failed";
```

Comment:
need to do this across lib. a wrapper around new should be implemented in this, could you please file a ticket for that and we can take it up.